bake/compose.go

```diff
@@ -70,7 +79,7 @@ func ParseCompose(dt []byte) (*Config, error) {
 				dockerfilePathP = &dockerfilePath
 			}
 			g.Targets = append(g.Targets, s.Name)
-			t := &Target{
+			err = mergo.Merge(t, &Target{
```

Comment:
There is a custom `Merge` method for targets.

Reply:
I've simplified the behavior. Should be ok now.